src/test/unit_tests/test_s3select.js

```diff
@@ -0,0 +1,216 @@
+    const input_stream = stream.Readable.from([input_str]);
+    const collect = new CollectChunks();
+    await stream.promises.pipeline(input_stream, s3select, collect);
+    return Buffer.concat(collect.chunks).toString();
+}
+
+async function simulate_async_loop(writable) {
+    writable.write("this is an ");
+    await new Promise(resolve => {
+        writable.write("input string.", null, () => {
+            resolve();
+        });
+    });
+    assert(writable.emit('error', new Error("I am an error event in an async loop.")));
+}
+
+
+mocha.describe('s3select', function() {
+
+    mocha.it('should create', function() {
+        const select_args = {
+            //query casts second column into integers, and sums them.
+            query: "select sum(int(_2)) from stdin;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const s3select = new s3select_utils.S3SelectStream(select_args);
+        s3select.destroy();
+    });
+
+    mocha.it('csv - select star', async function() {
+        const select_args = {
+            query: "select * from stdin;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const output = await run_sql(select_args, csv_small_str);
+        assert.strictEqual(output, csv_small_str, "wrong select output.");
+    });
+
+    mocha.it('csv - select sum with where', async function() {
+        const select_args = {
+            //sum every third number starting from row 1 (ie rows 1, 4, 7) in column 2
+            query: "select sum(int(_2)) from stdin where int(_1) % 3 = 1;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const output = await run_sql(select_args, csv_small_str);
+        assert.strictEqual(output, "1554", "wrong select output.");
+    });
+
+
+    mocha.it('csv - select with header name', async function() {
+        const select_args = {
+            //select column named 'c' where int value of column 'a' is less than 5
+            query: "select c from stdin where int(a) < 5;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n", "FileHeaderInfo": "USE"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const output = await run_sql(select_args, csv_small_str);
+        assert.strictEqual(output, "339\n741\n53\n233\n55\n", "wrong select output.");
+    });
+
+    mocha.it('json - select json', async function() {
```

Comment:
You mentioned json twice, maybe it can be: `'json - select column of integers fields'`